src/pal/src/exception/seh-unwind.cpp

```diff
@@ -107,12 +107,16 @@ static void UnwindContextToWinContext(unw_cursor_t *cursor, CONTEXT *winContext)
 
 static void GetContextPointer(unw_cursor_t *cursor, int reg, PDWORD64 *contextPointer)
 {
+#if defined(__APPLE__)
+    unw_get_reg(cursor, reg, (unw_word_t *) contextPointer);
```

Comment:
This is not correct. You are setting the context pointer to the register value. But it needs to be assigned a pointer to a stack location where the register is stored (and the assignment should happen only in case the register is stored on the stack at all). 
But I understand there is no API in the libunwind on OSX for that. I will write a standalone comment on the issue and possible ways to solve it.